### ISSUE TITLE
Rule Change: radix should warn when Number.parseInt is used as an argument in a map function

### ISSUE THREAD

**Issue by martinschayna** [eslint/eslint#15603]
### What rule do you want to change?

radix

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const numbers = ["22", "9", "60", "12", "4", "56"];

console.log(numbers.map(Number.parseInt));  

// output is [ 22, NaN, NaN, 5, NaN, NaN ]
```

### What does the rule currently do for this code?

Nothing

### What will the rule do after it's changed?

Array's method `map` expects closure as an parameter and pass three arguments to it, value, index and array. When `Number.parseInt` is used as an argument directly, it is error in every case, because the index of iteration is passed as second argument `radix` to it.

The rule should warn about it and suggest use of inline closure just like `array.map(x => Number.parseInt(x))`

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @martinschayna, thanks for the issue!

This is an interesting proposal, and it definitely makes sense to disallow the use of `parseInt`/`Number.parseInt` directly as  callbacks of `map` and other array methods.

However, this is already covered by a plugin rule [`unicorn/no-array-callback-reference`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md). It reports `parseInt`/`Number.parseInt` (I've just checked with your example) and much more, so I'm not sure if it's useful to update the core `radix` rule since there's already a more generic rule in a popular plugin.

**Comment by nzakas** [comment]
I also don’t think updating the radix rule for this case makes much sense vs a better already existing alternative in a plugin.
